SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -183,3 +246,67 @@ function Test-SecretVault
     )
     return (Get-Command lpass -ErrorAction SilentlyContinue) -and ((lpass status) -match "Logged in as .*")
 }
+
+function Get-SimpleSecret {
+    [CmdletBinding()]
+    param (
+        $Fields,
+        $Note
+    )
+    
+    $IsCredentials = $Fields.where( { $_.key -in @('Username', 'Password') }, 'first').count -eq 1
+    if ($IsCredentials) {
+        $username = $Fields.Where( { $_.key -eq 'Username' }, 'first') | Select-Object -ExpandProperty value
+        if ($null -eq $username) { $username = '' }
+        $password = $Fields.Where( { $_.key -eq 'Password' }, 'first') | Select-Object -ExpandProperty value
+        if ($null -eq $password) { $password = '' }
+        if ("" -ne $password) { $password = $password | ConvertTo-SecureString -AsPlainText -Force }
+        $output = [System.Management.Automation.PSCredential]::new($username, $password)
+    }
+    else {
+        if ($null -eq $Note) { $output = "" } else { $output = $Note }
+    }
+    return $output
+}
+
+function Get-ComplexSecret {
+    [CmdletBinding()]
+    param (
+        $Fields,
+        $Note,
+        $Raw
+    )
+    $Dupes = ($Fields | Group-Object key).Where( { $_.Count -gt 1 })
+    
+    if ($Dupes.count -gt 0) {
+        $Dupesstr = ($dupes | ForEach-Object { $_.Group.key -join ',' }) -join "`n"
+        
+        Write-Warning -Message @"
+The record contains multiple fields with the same name.
+$Dupesstr
+A Raw parameter will be added to the returned object with the raw object. Only the first field found with a duplicated name will be included in the parsed object.
+
+"@
+
+    }
+
+    $Output = @{}
+    if (![String]::IsNullOrEmpty($Note)) { 
+        $Output.Notes = $Note
+        $Fields = $Fields | Select-Object -SkipLast 1
```

Comment:
Notes is always last. Even if you create a custom type, add the Notes field first and add something afterward, it get slided down.